src/main/java/jenkins/branch/MultiBranchProjectFactory.java

```diff
@@ -46,21 +47,67 @@
  * how to get started using the type of project factory. This view is displayed when there are no subfolders found.
  *
  * @since 0.2-beta-5
+ * @see OrganizationFolder#getProjectFactories
  */
 public abstract class MultiBranchProjectFactory extends AbstractDescribableImpl<MultiBranchProjectFactory> implements ExtensionPoint {
 
     /**
-     * Creates a multibranch project for a given set of SCM sources if they seem compatible.
+     * Determines whether this factory recognizes a given configuration.
      * @param parent a folder
      * @param name a project name
      * @param scmSources a set of SCM sources as added by {@link jenkins.scm.api.SCMSourceObserver.ProjectObserver#addSource}
      * @param attributes a set of metadata attributes as added by {@link jenkins.scm.api.SCMSourceObserver.ProjectObserver#addAttribute}
      * @param listener a way of reporting progress
-     * @return a new uninitialized multibranch project (do not configure its {@link MultiBranchProject#getSourcesList} or call {@link MultiBranchProject#onCreatedFromScratch}), or null if unrecognized
+     * @return true if recognized
      */
+    public boolean recognizes(@Nonnull ItemGroup<?> parent, @Nonnull String name, @Nonnull List<? extends SCMSource> scmSources, @Nonnull Map<String,Object> attributes, @Nonnull TaskListener listener) throws IOException, InterruptedException {
+        if (Util.isOverridden(MultiBranchProjectFactory.class, getClass(), "createProject", ItemGroup.class, String.class, List.class, Map.class, TaskListener.class)) {
+            return createProject(parent, name, scmSources, attributes, listener) != null;
```

Comment:
It is defined below, and deprecated. It returns `MultiBranchProject` or null, and here I am checking which.